HTMLSerializer.js

```diff
@@ -347,6 +413,21 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Create a function that will generate unique strings which can be used as
+   * ids.
+   *
+   * @return {Function} A funtion that generates a unique string each time it is
+   *     called.
+   */
+  generateIDGenerator() {
```

Comment:
Either of your options sounds reasonable to me.

A third option:
It could be nice to keep around the original ids in a best-effort sort of way. I can imagine wanting to take a snapshot of this github page, for example, which has usefully named ids (js-pjax-loader-bar). To do this you'd just keep a list of ids you've generated and rename any new id attributes you encounter that collide.